src/ebay-star-rating-select/__tests__/index.spec.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react'
+import { render, fireEvent, cleanup } from '@testing-library/react';
+import { Isolated,Fieldset } from './mocks'
+
+const htmlSnap =(ui: React.ReactElement): void => {
+    const { asFragment } = render(ui);
+    expect(asFragment()).toMatchSnapshot();
+}
+
+let component;
+let onChangeSpy = jest.fn()
+let onFocusSpy = jest.fn()
+let onKeyDownSpy = jest.fn()
+
+describe('star-rating-select', () => {
+    it('renders defaults', async () => {
+        await htmlSnap(<Isolated a11yText={null} a11yStarText={null}/>);
+    });
+
+    it('renders basic fieldset', async () => {
+        await htmlSnap(<Fieldset/>);
+    });
+
+    it('renders with 0 selected', async () => {
+        await htmlSnap(<Isolated value="0"/>);
+    });
+
+    it('renders with 2 selected', async () => {
+        await htmlSnap(<Isolated value="2"/>);
+    });
+
+    it('renders with 5 selected', async () => {
+        await htmlSnap(<Isolated value="5"/>);
+    });
+
+    describe('given star rating', () => {
+        afterEach(cleanup);
+
+        beforeEach(async () => {
+            onChangeSpy.mockReset()
+            component = await render(<Isolated onChange={onChangeSpy} />);
+            await fireEvent.click(component.getByLabelText('2 stars'));
+        });
+
+        it('then it emits the event', () => {
+            const changeEvents = onChangeSpy.mock.calls;
+            expect(changeEvents.length).toBe(1);
+
+            const eventArgs = changeEvents[0];
+            expect(eventArgs.length).toBe(2);
+            expect(eventArgs[1]).toBe(2);
```

Comment:
You can use better assertions:

```js
  expect(onChangeSpy).toBeCalledTimes(1);
  expect(onChangeSpy).toBeCalledWith(expect.anything(), 2);  // since we only care about the 2nd param
```